tasks/base_tasks.py

```diff
@@ -1592,3 +1610,133 @@ def collect_meta_wrappers(test_module=None, test_all=True):
             yield t, params
             if not test_all:
                 break
+
+
+class CreateRepoTable(Task):
+    schema = 'repo'
+    table = 'filerepo'
+
+    def run(self):
+        session = current_session()
+
+        create_schema = '''
+                        CREATE SCHEMA IF NOT EXISTS "{schema}"
+                        '''.format(schema=self.output().schema)
+        session.execute(create_schema)
+
+        create_table = '''
+                        CREATE TABLE IF NOT EXISTS "{schema}".{table} (
+                            id TEXT,
+                            version NUMERIC,
+                            checksum TEXT,
+                            url TEXT,
+                            path TEXT,
+                            added TIMESTAMP,
+                            PRIMARY KEY(id, version)
+                        );
+                        '''.format(schema=self.output().schema,
+                                   table=self.output().tablename)
+        session.execute(create_table)
+        session.commit()
```

Comment:
Probably caused by my ignorance: I see a `commit`, but not an explicit transaction beginning, or error handling with a proper `rollback`. How come?

Reply:
The transaction is managed by SQLAlchemy (`session = current_session()`). But you are right, a `except: session.rollback()` won't hurt